sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/lang/SqlDelightQueriesFile.kt

```diff
@@ -107,14 +111,32 @@ class SqlDelightQueriesFile(
    * A collection of all the adapters needed for arguments or result columns in this query.
    */
   internal val requiredAdapters by lazy {
-    val argumentAdapters = PsiTreeUtil.findChildrenOfType(this, SqlBindExpr::class.java)
-      .mapNotNull { typeResolver.argumentType(it).parentAdapter() }
+    val binders = PsiTreeUtil.findChildrenOfType(this, SqlBindExpr::class.java)
+    val argumentAdapters = binders.singleQuestionMark() ?: binders.mapNotNull {
```

Comment:
this only works if theres only a single question mark in the whole file

Reply:
Thanks, added a test with two insert statements